### PR TITLE
Teach the server to choose the right node attestor

### LINKED ISSUE #222: SPIRE server should route Attestation requests to the right plugin
The server currently gives attestable data to the first node attestor plugin that the catalog returns. In reality, we should be choosing the plugin which matches the type of attestable data we are passing. Make it so

### PR DESCRIPTION
**Pull request by evan2645**
Fixes #222 

@mlakewood

### CODE REVIEW

**Comment by evan2645** [comment]
Oops, I meant @juturnas not Mark :) sorry!

File: pkg/server/node.go [hunk 1]
```diff
@@ -250,8 +250,15 @@ func (s *nodeServer) attest(
 	attestedData *common.AttestedData, attestedBefore bool) (
 	response *nodeattestor.AttestResponse, err error) {
 
-	// TODO: Pick the right node attestor [#222]
-	nodeAttestor := s.catalog.NodeAttestors()[0]
+	// Pick the right node attestor
+	var nodeAttestor nodeattestor.NodeAttestor
+	for _, a := range s.catalog.NodeAttestors() {
+		mp := s.catalog.Find(a)
+		if mp.Config.PluginName == attestedData.Type {
```
Comment (kunzimariano) [inline pkg/server/node.go]: Shouldn't we consider the situation where it doesn't find a plugin?

Comment (evan2645) [reply pkg/server/node.go]: oh.... lol yea! Will fix :) thanks!

**Comment by evan2645** [comment]
@kunzimariano fixed, please re-review

File: pkg/server/node.go [hunk 2]
```diff
@@ -250,8 +250,19 @@ func (s *nodeServer) attest(
...
```
Comment (walmav) [inline pkg/server/node.go]: Does this imply attestedData Type need to match the pluginName

Comment (evan2645) [reply pkg/server/node.go]: Yes... `Type` is probably a bad choice for name here, but changing the name of the field would be cross-cutting. Opened https://github.com/spiffe/spire/issues/289 to track, feel free to add comments there

**Pull request merged**